### PR TITLE
gdrive: init at 2.1.0

### PR DESCRIPTION
**Pull request by rzetterberg**
###### Motivation for this change

Needed to upload some stuff to Google Drive, didn't find gdrive in nixpkgs,
created a package, installed it, it worked, thought "why not add the package?".

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [X] NixOS (17.09pre102667.2839b10)
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

This package should work in macOS too, but I have not tested it on macOS, therefor I added `platforms.linux`

Running this script in the root of the nixpkgs project:

```bash
#!/usr/bin/env bash

set -e

nixpkgs_root=$(pwd)
tmp_shell_file=$(mktemp)

cat <<EOF > "$tmp_shell_file"
{ system ? builtins.currentSystem }:

let
  pkgs = import <nixpkgs> { inherit system; };
in
  pkgs.callPackage "$nixpkgs_root/pkgs/applications/networking/gdrive/default.nix" {
    inherit (pkgs);
  }
EOF

nix-env -i -f "$tmp_shell_file"

gdrive version

rm "$tmp_shell_file"
```

Should result in the following output:

```
installing ‘gdrive-2.1.0’
building path(s) ‘/nix/store/fa7vcac8fgg6dg2bgk3kwm7n9m2gggry-user-environment’
created 1125 symlinks in user environment
gdrive: 2.1.0
Golang: go1.7.4
OS/Arch: linux/amd64
```

I tested running the list command against a newly created google account. After
authenticating it worked:

2017-04-12-184410_1364x729_scrot

I have also successfully uploaded files.

### CODE REVIEW

File: pkgs/applications/networking/gdrive/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    repo   = "gdrive";
+    sha256 = "0ywm4gdmrqzb1a99vg66a641r74p7lglavcpgkm6cc2gdwzjjfg7";
+    inherit rev;
+  };
+
+  meta = with stdenv.lib; {
+    homepage    = https://github.com/prasmussen/gdrive;
+    description = "A command line utility for interacting with Google Drive";
+    platforms   = platforms.linux;
+    license     = licenses.mit;
```
Comment (Mic92) [inline pkgs/applications/networking/gdrive/default.nix]: Do you want to maintain this package?

Comment (rzetterberg) [inline pkgs/applications/networking/gdrive/default.nix]: @Mic92 Yes! I'm not part of `nixpkgs/lib/maintainers.nix`, should I add myself in a separate PR?

Comment (Mic92) [inline pkgs/applications/networking/gdrive/default.nix]: Just do it in this one.

Comment (bjornfor) [inline pkgs/applications/networking/gdrive/default.nix]: IMHO, preferably in a separate commit. So commit 1: add lib/maintainers.entry, commit 2: add package using your new entry in maintainers.nix.

Reasoning: sometimes when backporting to the stable branch, it's nice if the maintainers entry is decoupled from other changes.

**Comment by rzetterberg** [comment]
Made changes suggested from review of @Mic92 and @bjornfor.

After I made the changes I ran:

```
$ nix-env -e gdrive
$ nix-env -i gdrive -f .
... truncated lines
lzjf-bash-4.4-p12/bin/bash"
/nix/store/936mlgszb7spsq4sf5ik1zxahfd7jy01-gdrive-2.1.0/share/go/src/github.com/prasmussen/gdrive/_release/upload.sh: interpreter directive changed from "/usr/local/bin/bash" to "/nix/store/fibk8jc2iwswn1q1d7aw98pcw2l3lzjf-bash-4.4-p12/bin/bash"
building path(s) ‘/nix/store/88csbvf519rv816r211bgjqwghd3xx2q-user-environment’
created 1135 symlinks in user environment

$ gdrive version
gdrive: 2.1.0
Golang: go1.8
OS/Arch: linux/amd64
```

**Pull request merged**